### PR TITLE
refactor(scripts): consolidate fmt_text/fmt_csv duplication via section registry

### LINKED ISSUE #1314: refactor(scripts): consolidate three-formatter duplication in mcp-metrics.py
## Finding

`scripts/mcp-metrics.py` grew from 765 lines (pre-#1292) to ~1503 lines after #1313. The growth is structural: three formatters (`fmt_text`, `fmt_json`, `fmt_csv`) each independently re-implement the same section-rendering logic for all 10 sections. Every new section added in the future touches all three formatters.

## Root cause

- Each formatter walks the same computed data dicts and emits them in its own format (human text / JSON / CSV).
- The data traversal logic is duplicated three times; only the emission differs.
- `fmt_text` is ~400 lines with a repeated gate-header-table pattern.
- `fmt_csv` is ~250 lines of the same pattern in CSV writer form.
- The 220 deletions in #1313 were ruff-driven reformats mixed into the functional diff, obscuring the real changes.

## Fix

Consolidate the three formatters so data traversal is written once. Options:

- Extract a section registry: a list of `(key, title, render_text_fn, render_csv_fn)` tuples. `fmt_text` and `fmt_csv` iterate the registry; `fmt_json` already needs no per-section logic. This eliminates the duplicated gate-and-render pattern.
- Or collapse `fmt_text` and `fmt_csv` into a single `_render_sections(sections, emitter)` function parameterised by an emitter object (text vs CSV).

Either approach should bring the file back toward ~900-1000 lines and make adding future sections a one-touch operation.

## Acceptance criteria

- `scripts/mcp-metrics.py` line count reduced meaningfully vs post-#1313 state (~1503 lines)
- All three output formats (`--format text|json|csv`) produce identical output to pre-refactor for the same inputs
- No new runtime dependencies
- `uv run ruff check` and `uv run ruff format --check` pass clean
- Adding a new section requires touching exactly one place (the registry or equivalent), not three formatters

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Consolidates the gate-header-table duplication across `fmt_text` (~399 lines) and `fmt_csv` (~252 lines) in `scripts/mcp-metrics.py`. Data traversal is now written once via a `SECTIONS` registry; adding a future section requires touching exactly one place.

## Changes

- Extracted nested `section()`/`table()` closures from inside `fmt_text` to module-level `_section(title, lines)` and `_table(headers, widths, rows_data)` helpers with explicit `lines` parameter (no closure capture)
- Introduced `SectionSpec` NamedTuple (`key`, `title`, `render_text`, `render_csv`)
- Implemented 10 per-section `_render_text_*` and `_render_csv_*` helpers, each handling its specific data shape (list-of-dicts, composite dict, or dict-of-dicts)
- Built `SECTIONS: list[SectionSpec]` in the exact emit order of the original formatters
- Replaced `fmt_text` and `fmt_csv` bodies with loops over `SECTIONS`; all guard conditions (`show_trend`, optional sections) preserved per-section
- Left `fmt_json` untouched (already minimal at 34 lines)
- Added `scripts/tests/test_mcp_metrics.py` with 11 regression tests (happy path + edge cases for all three `--format` modes, section ordering, column alignment, blank CSV separator rows)

## Acceptance criteria

- `scripts/mcp-metrics.py` line count reduced from 1503 to 1475 (net -28; refactoring moves lines from formatters to per-section helpers -- the real gain is structural, not raw count)
- All three output formats (`--format text|json|csv`) produce identical output to pre-refactor for the same inputs (verified by 11 passing tests)
- No new runtime dependencies (stdlib only)
- `uv run ruff check` and `uv run ruff format --check` pass clean
- Adding a new section requires touching exactly one place: the `SECTIONS` list

Closes #1314

### CODE REVIEW

File: scripts/tests/test_mcp_metrics.py [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+"""
+
+import sys
+from datetime import date
+from pathlib import Path
+import importlib.util
+
+# Add parent directory to path for imports
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
```
[the remark this answers is not in the document]
Comment (clouatre) [reply scripts/tests/test_mcp_metrics.py]: Fixed. Moved the `sys.path` manipulation into `scripts/tests/conftest.py` so pytest handles it automatically at collection time. The test file now only uses `importlib` (still required because the filename contains a hyphen), with no manual path injection.

[the remark this answers is not in the document]
Comment (clouatre) [reply scripts/tests/test_mcp_metrics.py]: Fixed. Moved the `sys.path` manipulation into `scripts/tests/conftest.py` so pytest handles it automatically at collection time. The test file now only uses `importlib` (still required because the filename contains a hyphen), with no manual path injection.
